reaktive/src/commonMain/kotlin/com/badoo/reaktive/completable/Merge.kt

```diff
@@ -33,13 +40,9 @@ fun Iterable<Completable>.merge(): Completable =
                 }
             )
         }
-
-        if (activeSourceCount.addAndGet(-1) == 0) {
```

Comment:
This change will brake the operator in case when one of the upstreams will complete while subscribing to others. The following test demonstrates the issue, could you please add it as well? The `toList` conversion in this particular operator is therefore unnecessary.

```kotlin
    @Test
    fun does_not_complete_WHEN_one_upstream_completed_while_subscribing_to_others() {
        val observer =
            merge(
                completableUnsafe { observer ->
                    observer.onSubscribe(Disposable())
                    observer.onComplete()
                },
                TestCompletable()
            ).test()

        observer.assertNotComplete()
    }
```

Reply:
Yes, i will add this test. My bad, didn't test this operator before fix.

Reply:
I think that `completableOfEmpty` is better due less code, check for `isDisposed` inside `completableOfEmpty` can't affect this, right?

Reply:
We are trying to avoid unrelated production code in tests. Because we wan't to test exactly what is under test, and not other (unrelated) code. If `observableOfEmpty` at some point will be broken then this particular test will be red, which ideally should not happen.

Reply:
Would you mind to use `completableUnsafe` instead? After that I think we are done.

Reply:
Ok.